virttest/utils_cpi.py

```diff
@@ -0,0 +1,937 @@
+                self.cpi_data[field_name] = None
+
+        return self.cpi_data
+
+    def get_guest_distro_info(self):
+        """
+        Get distribution information from guest VM using utils_misc function
+
+        :return: Dictionary with guest distribution information
+        """
+        guest_distro_info = utils_misc.get_guest_distro_info(
+            self.vm, serial=self.serial
+        )
+        self.guest_distro_info = guest_distro_info
+        return guest_distro_info
+
+    def check_system_type(self):
+        """
+        Check that system_type matches the configured value
+
+        :return: True if check passes
+        :raises: AssertionError if check fails
+        """
+        if "system_type" not in self.cpi_data:
+            raise RuntimeError("system_type not available in CPI data")
+
+        system_type = self.cpi_data["system_type"]
+
+        try:
+            config = self.get_cpi_config()
+            expected_system_type = config.get("CPI_SYSTEM_TYPE", "")
+        except Exception as e:
+            logging.warning(f"Could not read CPI config, using empty string: {e}")
+            expected_system_type = ""
+
+        if system_type == "" and expected_system_type != "":
+            raise AssertionError(
+                f"system_type is empty (Secure Execution guest without CPI_PERMIT_ON_PVGUEST='1'?). "
+                f"Expected: {expected_system_type}"
+            )
+
+        if system_type != expected_system_type:
+            raise AssertionError(
+                f"system_type mismatch: expected '{expected_system_type}', got '{system_type}'"
+            )
+
+        logging.info(f"system_type check passed: {system_type}")
+        return True
+
+    def check_system_name(self):
+        """
+        Check that system_name matches the configured value
+
+        :return: True if check passes
+        :raises: AssertionError if check fails
+        """
+        if "system_name" not in self.cpi_data:
+            raise RuntimeError("system_name not available in CPI data")
+
+        system_name = self.cpi_data["system_name"]
+
+        try:
+            config = self.get_cpi_config()
+            expected_system_name = config.get("CPI_SYSTEM_NAME", "")
+        except Exception as e:
+            logging.warning(f"Could not read CPI config, using empty string: {e}")
+            expected_system_name = ""
+
+        if system_name == "" and expected_system_name != "":
+            raise AssertionError(
+                f"system_name is empty (Secure Execution guest without CPI_PERMIT_ON_PVGUEST='1'?). "
+                f"Expected: {expected_system_name}"
+            )
+
+        if system_name != expected_system_name:
+            raise AssertionError(
+                f"system_name mismatch: expected '{expected_system_name}', got '{system_name}'"
+            )
+
+        logging.info(f"system_name check passed: {system_name}")
+        return True
+
+    def check_sysplex_name(self):
```

Comment:
check_sysplex_name has some similar logic with check_system_name and check_system_type, I think we may have public func for them.